dask/dataframe/io/io.py

```diff
@@ -524,7 +524,7 @@ def from_delayed(dfs, meta=None, divisions=None, prefix='from-delayed'):
                             type(df).__name__)
 
     if meta is None:
-        meta = dfs[0].compute()
+        meta = dask.delayed(make_meta)(dfs[0]).compute()
```

Comment:
+1 on this - but is there are need to call `make_meta` again on it on the next like (I expect it's a no-op)?

Reply:
I've gone ahead and added an `else:` to only call this in the other side of the branch

Reply:
Thanks